proposals/XXXX-atf-selenium-support_review.md

```diff
@@ -0,0 +1,236 @@
+
+# ATF Selenium Support
+
+* Proposal: [SDL-0340](0340-atf-selenium-support.md)
+* Author: [Jack Byrne](https://github.com/JackLivio)
+* Status: **In Review**
+* Impacted Platforms: [ATF / ATF Test Scripts / Generic HMI]
+
+## Introduction
+
+This proposal is to integrate the Generic HMI into ATF tests scripts by using Selenium to control a web browser programmatically.
+
+## Motivation
+
+Currently ATF is used to test SDL Core as a black box. ATF uses simulated mobile applications and a simulated HMI to test SDL Core functionality. While ATF is great at testing SDL Core, the author aims to create a solution for having an automated test suite that tests both SDL Core and the Generic HMI together.
+
+## Proposed solution
+
+The solution is to create a python Selenium WebDriver that will control actions and validate the content of the HMI when it is interacting with SDL Core.
+
+### Selenium WebDriver Application
+
+#### Life Cycle 
+
+ATF will launch the python app using the os.execute() method. The python application will then launch a physical instance of the Generic HMI in the chromium browser.
+
+```
+from selenium import webdriver
+driver = webdriver.Chrome()
+driver.get(GENERIC_HMI_LOCATION)
+```
+
+This application will stay running until the end of the test when ATF will shut down the process. To close the browser the Selenium driver calls:
+
+```
+driver.close()
+```
+
+#### WebDriver Actions
+
+Selenium is capable of finding elements on a page and interacting with them.
+
+```
+driver.find_element_by_id(ELEMENT_ID).click()
+```
+
+This type of call will be used to press different buttons in the HMI. This will be useful for activating an app, opening the menu, pressing a softbutton etc. These actions will trigger the HMI's state to change and also trigger applicable RPC messages to be sent back to ATF for validation.
+
+#### WebDriver Validation
+
+The Selenium WebDriver can also be used to validate the content on the page. For example if a Show request is sent to the HMI, the web driver should check that the text is correctly displayed.
+
+```
+return driver.find_element_by_id(ELEMENT_ID).text() == test.text
+```
+
+#### Communication with ATF
+
+This python application will have a websocket server that will receive messages from ATF for specific test actions and content validation.
+
+### Extra Websocket Server and Clients in ATF
+
+#### Selenium Websocket Client
+
+ATF will need to create a websocket client that relays messages to the python Selenium application. These messages will tell the WebDriver to click buttons and verify content displayed on the screen.
+
+#### HMI Websocket Server
+
+ATF will need to create a websocket server that will receive messages from the HMI. This server will then be routed through the ATF test logic to verify the contents of the HMI messages. After verifying that the messages are valid, ATF will pass the HMI message to SDL Core.
```

Comment:
I think the typo is here because a websocket server is opened on python app side. So ATF should create websocket client and connect to websocket server of python HMI testing app.